src/data/repositories/UserD2ApiRepository.ts

```diff
@@ -61,8 +63,9 @@ export class UserD2ApiRepository implements UserRepository {
             created: new Date(user.created),
             userGroups: user.userGroups,
             username: user.userCredentials.username,
+            apiUrl: `${this.api.baseUrl}/api/users/${user.id}.json`,
             userRoles: user.userCredentials.userRoles,
-            lastLogin: new Date(user.userCredentials.lastLogin),
+            lastLogin: lastLogin === undefined ? "N/A" : new Date(lastLogin),
```

Comment:
Instead of setting here ``N/A`` for undefined values, this should be handled on the ``presentation`` layer in a component scope.